src/MUIPlacesAutocomplete.jsx

```diff
@@ -1,6 +1,213 @@
+          ...other,
+        }}
+      />
+    )
+  }
+
+  static renderSuggestionsContainer({ containerProps, children }) {
+    return (
+      <div {...containerProps}>
+        {children}
+        {children ? (
+          <div style={{ display: 'flex' }}>
+            <span style={{ flex: 1 }} />
+            <img src={googleLogo} alt="" />
+          </div>
+        ) : null}
+      </div>
+    )
+  }
+
+  // Renders suggestions where they are highlighted based on the parts of the suggestion that match
+  // the query the user entered. This is inline with the Google Maps webapp at the time of writing.
+  // This behavior is opposite of how the Google Search bar/component/element works though.
+  static renderSuggestion({ description }, { query, isHighlighted }) {
+    // Calculate the chars to highlight in the suggestion 'description' based on the query that the
+    // user provided us. An array is returned and if any chars ought to be highlighted the array
+    // will contain a pair ([a, b]) which denote the indexes of chars to highlight (i.e.
+    // text.slice(a, b)).
+    const matches = match(description, query)
+
+    // Break up the suggestion 'description' based on the parts that matched. An array is returned
+    // of parts where each one has an indication of if it ought to be highlighted or not.
+    const parts = parse(description, matches)
+
+    return (
+      <MenuItem selected={isHighlighted} component="div">
+        <div>
+          {parts.map((part, index) => {
+            if (part.highlight) {
+              // Since we are further breaking down an array there is nothing unique about the
+              // elements in the resulting array so we can disable the react/no-array-index-key
+              // ESLint rule when rendering our suggestion
+              // eslint-disable-next-line react/no-array-index-key
+              return <strong key={index} style={{ fontWeight: 500 }}>{part.text}</strong>
+            }
+
+            // Since we are further breaking down an array there is nothing unique about the
+            // elements in the resulting array so we can disable the react/no-array-index-key
+            // ESLint rule when rendering our suggestion
+            // eslint-disable-next-line react/no-array-index-key
+            return <span key={index} style={{ fontWeight: 300 }}>{part.text}</span>
+          })}
+        </div>
+      </MenuItem>
+    )
+  }
+
+  constructor() {
+    super()
+
+    // Control the <input> element/<Autosuggest> component and make this React component the source
+    // of truth for their state.
+    this.state = { value: '', suggestions: [] }
+
+    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this)
+    this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this)
+    this.onChange = this.onChange.bind(this)
+  }
+
+  componentWillMount() {
+    this.autocompleteService = new window.google.maps.places.AutocompleteService()
```

Comment:
Can you give guidance for how to properly consume 3rd party libraries that are expected to be loaded in the browser/window?

For example should I account for cases where the library doesn't load and if so what ought I do? Or is this fine the way it is implemented and if the library doesn't load its okay to let this component blow up and errors show up in the console?

Reply:
Ya. This seems like a different set of problems that I have never had to consider before.

I saw that package (and others) in the past when I was looking for something myself but thought that they weren't sufficient in terms of getting a MUI feeling applied to them. Maybe I was wrong just based on my limited knowledge.

Not looking to re-invent the wheel but it feels good to produce this myself. I just get to further learn more about the ecosystem Im working with. Plus I have open sourced something! w00t! :)

Reply:
https://www.npmjs.com/package/google-maps

there ya go, someone did simply wrap it in a module with 0 changes, should check it out

Reply:
otherwise, yeah, all good

Reply:
TY. Im going to make a task to look at this.